File: rb/features.rb
# FriedApps SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module FriedAppsFeatures
  def self.make_feature(name)
    case name
    when "base"
      FriedAppsBaseFeature.new
    when "test"
      FriedAppsTestFeature.new
    else
      FriedAppsBaseFeature.new
    end
  end
end

File: py/features.py
# FriedApps SDK feature factory

from feature.base_feature import FriedAppsBaseFeature
from feature.test_feature import FriedAppsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FriedAppsBaseFeature(),
        "test": lambda: FriedAppsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
